sdk/resourcemanager/azure-resourcemanager-network/src/test/java/com/azure/resourcemanager/network/NetworkInterfaceOperationsTests.java

```diff
@@ -509,6 +511,113 @@ public void canAssociateNatGateway() {
+            .create();
+
+        nic.refresh();
+        List<NetworkInterfaceIpConfigurationInner> ipConfigurationInnerList =  nic.innerModel().ipConfigurations()
+            .stream().filter(ipConfig -> DeleteOptions.DELETE.equals(ipConfig.publicIpAddress().deleteOption()))
+            .collect(Collectors.toList());
+
+        Assertions.assertEquals(2, ipConfigurationInnerList.size());
+    }
+
+    @Test
+    public void canUpdateNicWithMultipleDeleteOptions() {
+        String subnetName = generateRandomResourceName("subnet-", 15);
+        resourceManager.resourceGroups().define(rgName).withRegion(Region.US_EAST).create();
+        Network vnet = networkManager.networks()
+            .define(generateRandomResourceName("vnet-", 15))
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+            .withAddressSpace("10.0.0.0/28")
+            .withSubnet(subnetName, "10.0.0.0/28")
+            .create();
+
+        NetworkInterface nic = networkManager.networkInterfaces()
+            .define(generateRandomResourceName("nic-", 15))
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+            .withExistingPrimaryNetwork(vnet)
+            .withSubnet(subnetName)
+            .withPrimaryPrivateIPAddressDynamic()
+            .withNewPrimaryPublicIPAddress()
+            .withPrimaryPublicIPAddressDeleteOptions(DeleteOptions.DELETE)
+            .defineSecondaryIPConfiguration("secondary1")
+                .withExistingNetwork(vnet)
+                .withSubnet(subnetName)
+                .withPrivateIpAddressDynamic()
+                .withNewPublicIpAddress()
+                .withPublicIPAddressDeleteOptions(DeleteOptions.DETACH)
+                .attach()
+            .defineSecondaryIPConfiguration("secondary2")
+                .withExistingNetwork(vnet)
+                .withSubnet(subnetName)
+                .withPrivateIpAddressDynamic()
+                .withNewPublicIpAddress()
+                .withPublicIPAddressDeleteOptions(DeleteOptions.DETACH)
+                .attach()
+            .create();
+
+        nic.refresh();
+        nic.update().withNewPrimaryPublicIPAddress().withPrimaryPublicIPAddressDeleteOptions(DeleteOptions.DETACH).apply();
+        nic.refresh();
+        List<NetworkInterfaceIpConfigurationInner> ipConfigurationInnerList = nic.innerModel().ipConfigurations()
+            .stream().filter(ipConfigurationInner -> DeleteOptions.DETACH.equals(ipConfigurationInner.publicIpAddress().deleteOption()))
+            .collect(Collectors.toList());
+        Assertions.assertEquals(3, ipConfigurationInnerList.size());
+
+        nic.update()
+            .withPrimaryPublicIPAddressDeleteOptions(DeleteOptions.DELETE)
+            .updateIPConfiguration("secondary1")
+            .withNewPublicIpAddress()
+            .withPublicIPAddressDeleteOptions(DeleteOptions.DELETE)
```

Comment:
same as https://github.com/Azure/azure-sdk-for-java/pull/39096/files#r1538788740